extension/src/popup/components/accountHistory/TransactionDetail/index.tsx

```diff
@@ -0,0 +1,161 @@
+import StellarSdk from "stellar-sdk";
+
+import { PopupWrapper } from "popup/basics/PopupWrapper";
+
+import { SubviewHeader } from "popup/components/SubviewHeader";
+
+import { emitMetric } from "helpers/metrics";
+import { openTab } from "popup/helpers/navigate";
+import { stroopToXlm } from "helpers/stellar";
+import { getIconUrlFromIssuer } from "@shared/api/helpers/getIconUrlFromIssuer";
+
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+
+import { METRIC_NAMES } from "popup/constants/metricsNames";
+
+import { HorizonOperation } from "@shared/api/types";
+import StellarLogo from "popup/assets/stellar-logo.png";
+
+import "./styles.scss";
+
+export interface TransactionDetailProps {
+  operation: HorizonOperation;
+  headerTitle: string;
+  isRecipient: boolean;
+  operationText: string;
+  externalUrl: string;
+  setIsDetailViewShowing: (isDetailViewShoing: boolean) => void;
+}
+
+export const TransactionDetail = ({
+  operation,
+  headerTitle,
+  isRecipient,
+  operationText,
+  externalUrl,
+  setIsDetailViewShowing,
+}: TransactionDetailProps) => {
+  const {
+    asset_code: assetCode,
+    asset_issuer: assetIssuer,
+    asset_type: assetType,
+    from,
+    created_at: createdAt,
+    transaction_attr: { fee_charged: feeCharged, memo },
+  } = operation;
+  const createdAtDateInstance = new Date(Date.parse(createdAt));
+  const createdAtLocalStrArr = createdAtDateInstance
+    .toLocaleString()
+    .split(" ");
+  const createdAtTime = `${createdAtLocalStrArr[1]
+    .split(":")
+    .slice(0, 2)
+    .join(":")} ${createdAtLocalStrArr[2]}`;
+  const createdAtDateStr = createdAtDateInstance
+    .toDateString()
+    .split(" ")
+    .slice(1)
+    .join(" ");
+
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const [networkIconUrl, setNetworkIconUrl] = useState("");
+  const [networkDomain, setNetworkDomain] = useState("");
+
+  useEffect(() => {
+    const fetchIconUrl = async () => {
+      const { networkUrl } = networkDetails;
+      const server = new StellarSdk.Server(networkUrl);
+      let iconUrl = "";
+      let assetDomain = "";
+
+      // TODO: Combine these 2 into 1 call. getIconUrlFromIssuer load's the issuer account from Horizon.
+      // Find a way to get the icon and the home domain in one call even when icon is cached
+      try {
+        ({ home_domain: assetDomain } = await server.loadAccount(assetIssuer));
+      } catch (e) {
+        console.error(e);
+      }
+      setNetworkDomain(assetDomain || " ");
+
+      try {
+        iconUrl = await getIconUrlFromIssuer({
+          key: assetIssuer || "",
+          code: assetCode || "",
+          networkDetails,
+        });
+      } catch (e) {
+        console.error(e);
+      }
+
+      setNetworkIconUrl(iconUrl);
+    };
+
+    if (assetIssuer) {
+      fetchIconUrl();
+    }
+  }, [assetCode, assetIssuer, isRecipient, networkDetails]);
+
+  return assetIssuer && !networkDomain ? null : (
```

Comment:
Yup, good question. In the case of a non-XLM payment, it takes a second to hit Horizon and get the `networkDomain` of the asset. This was causing an annoying UX where it would show the rest of the detail data, and then the `networkDomain` would flash in. 

This is a workaround that's basically saying, wait til we have *all* the data before we show anything 

Ideally, in the future, we will be caching not only the asset icon, but its `networkDomain` as well so we won't need to hit Horizon at all. We will already have all the data available to us. The TODO here will take care of that for us.
https://github.com/stellar/freighter/pull/399/files#diff-dacdc62063df120813b20d41a256cb1a25bb61b226443a9c6cbd1225bedc4023R74

Reply:
got it, thanks for explaining!